lib/chef/provider/service/macosx.rb

```diff
@@ -133,20 +149,32 @@ def disable_service
           unless @current_resource.enabled
             Chef::Log.debug("#{@new_resource} not enabled, not disabling")
           else
-            shell_out!(
-              "launchctl unload -w '#{@plist}'",
-              :user => @owner_uid, :group => @owner_gid
-            )
+            unload_service
           end
         end
 
+       def load_service
+          cmd = 'launchctl load -w '
+          cmd << "-S #{@session_type} " if @session_type
+          cmd << @plist
+          cmd = "#{@base_user_cmd} '#{cmd}'" if @console_user
+          shell_out_with_systems_locale!(cmd)
+        end
+
+        def unload_service
+          cmd = 'launchctl unload -w '
+          cmd << "-S #{@session_type} " if @session_type
+          cmd << @plist
+          cmd = "#{@base_user_cmd} '#{cmd}'" if @console_user
+          shell_out_with_systems_locale!(cmd)
+        end
+
         def set_service_status
           return if @plist == nil or @service_label.to_s.empty?
 
-          cmd = shell_out(
-            "launchctl list #{@service_label}",
-            :user => @owner_uid, :group => @owner_gid
-          )
+          cmd = "launchctl list #{@service_label}"
+          cmd = "#{@base_user_cmd} '#{cmd}'" if @console_user
+          cmd = shell_out_with_systems_locale(cmd)
```

Comment:
eh, cmd/res are fine. 

It's just that it looks like

1) set cmd to something
2) set cmd to something else
3) set cmd to yet something else

When you look at the first line, you're like, "okay cmd is a string that we're going to run" then you look down at new lines 179 and you think "wait, cmd.exitstatus? I thought it was a string we were running"

At the least, different variable names would be more clear.